numpy/lib/tests/test_function_base.py

```diff
@@ -3104,6 +3108,58 @@ def test_no_p_overwrite(self):
         np.quantile(np.arange(100.), p, interpolation="midpoint")
         assert_array_equal(p, p0)
 
+    def test_quantile_monotonic(self):
+        # GH 14685
+        # test that the return value of quantile is monotonic if p0 is ordered
+        p0 = np.arange(0, 1, 0.01)
+        quantile = np.quantile(np.array([0, 1, 1, 2, 2, 3, 3, 4, 5, 5, 1, 1, 9, 9, 9, 8, 8, 7]) * 0.1, p0)
+        equals_sorted = np.sort(quantile) == quantile
+        assert equals_sorted.all()
+
+    @hypothesis.given(arr=arrays(dtype=np.float, shape=st.integers(min_value=3, max_value=1000),
+                                 elements=st.floats(allow_infinity=False, allow_nan=False)))
+    def test_quantile_monotonic_hypo(self, arr):
+        p0 = np.arange(0, 1, 0.01)
+        quantile = np.quantile(arr, p0)
+        equals_sorted = np.sort(quantile) == quantile
+        assert equals_sorted.all()
+
+
+class TestLerp:
+    @hypothesis.given(t0=st.floats(allow_nan=False, allow_infinity=False,
+                                   min_value=0, max_value=1),
+                      t1=st.floats(allow_nan=False, allow_infinity=False,
+                                   min_value=0, max_value=1),
+                      a=st.floats(allow_nan=False, allow_infinity=False,
+                                  width=32),
+                      b= st.floats(allow_nan=False, allow_infinity=False,
+                                   width=32))
```

Comment:
These need to be 64 for this test to be interesting.

Reply:
that leads to overflows:

```
b=1.7976931348623157e+308
a=-1.7976931348623157e+308
np.subtract(b, a)
*** RuntimeWarning: overflow encountered in subtract
```

Reply:
That's potentially a problem, the previous implementation did not fail like that for those cases.

For now, can you do something like clip a and b to reasonable magnitudes? We still want to test the precision of 64-bit, even if we decide we don't care about the range.

Reply:
I changed the range from `-1e300` up to `1e300`, seems to work fine.

Are you sure that it wasn't a problem before? The overflow happens in the basic `add` and subtract methods and should be reproducible for specific combinations for the former implementation aswell since we are still using those methods.